# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/base/AppShutdown.cpp
# Commit: a727a8fe08f5
# Full Hash: a727a8fe08f50746757656ba9599047dac6c0420
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-02-13 21:42:57
# Regressor Bug: 1607505
# File Overlap Count: 3
# Description:
#   Bug 1607505 - Start writing StartupCache earlier in shutdown r=froydnj a=reland CLOSED TREE
#   
#   The initial thought for getting the StartupCache out of the shutdown
#   path was to simply not write it during shutdown, as it should write
#   60 seconds after startup, and the theory was that if the user shut
# ==============================================================================

diff -r 752a464c8ec8 -r a727a8fe08f5 xpcom/base/AppShutdown.cpp
--- a/xpcom/base/AppShutdown.cpp	Thu Feb 13 16:56:44 2020 +0000
+++ b/xpcom/base/AppShutdown.cpp	Wed Feb 12 19:02:12 2020 +0000
@@ -16,6 +16,7 @@
 #include "mozilla/CmdLineAndEnvUtils.h"
 #include "mozilla/PoisonIOInterposer.h"
 #include "mozilla/Printf.h"
+#include "mozilla/scache/StartupCache.h"
 #include "mozilla/StartupTimeline.h"
 #include "mozilla/StaticPrefs_toolkit.h"
 #include "mozilla/LateWriteChecks.h"
@@ -125,6 +126,8 @@
   int32_t lateWriteChecksPref =
       StaticPrefs::toolkit_shutdown_lateWriteChecksStage();
   sLateWriteChecksPhase = GetShutdownPhaseFromPrefValue(lateWriteChecksPref);
+
+  scache::StartupCache::GetSingleton()->MaybeInitShutdownWrite();
 }
 
 void AppShutdown::MaybeFastShutdown(ShutdownPhase aPhase) {
